File: 0001-two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const n = nums.length;
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (i !== j && nums[i] + nums[j] === target) return [i, j];
    }
  }
  return [n - 1, n - 1];
}

File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> mp;
        for(int i = 0; i < n; i++) {
            int prev = target - nums[i];
            if(mp.contains(prev)) return {mp[prev], i};
            mp[nums[i]] = i;
        }
        return {};
    }
};
